text/0089-metrics-views-api.md

```diff
@@ -0,0 +1,264 @@
+| OpenCensus | LastValue                                                                                          | Observer             | LastValue         |
+| Datadog    | [COUNT](https://docs.datadoghq.com/developers/metrics/types/?tab=count#metric-types)               | Counter              | Sum               |
+| Datadog    | [RATE](https://docs.datadoghq.com/developers/metrics/types/?tab=rate#metric-types)                 | Counter              | Rate\*            |
+| Datadog    | [GAUGE](https://docs.datadoghq.com/developers/metrics/types/?tab=gauge#metric-types)               | Observer             | LastValue         |
+| Datadog    | [HISTOGRAM](https://docs.datadoghq.com/developers/metrics/types/?tab=histogram#metric-types)       | Measure              | Histogram         |
+| Datadog    | [DISTRIBUTION](https://docs.datadoghq.com/developers/metrics/types/?tab=distribution#metric-types) | Measure              | Sketch\* or Exact |
+| Prometheus | [Counter](https://prometheus.io/docs/concepts/metric_types/#counter)                               | Counter              | Sum               |
+| Prometheus | [Gauge](https://prometheus.io/docs/concepts/metric_types/#gauge)                                   | Observer             | LastValue         |
+| Prometheus | [Histogram](https://prometheus.io/docs/concepts/metric_types/#histogram)                           | Measure              | Histogram         |
+| Prometheus | [Summary](https://prometheus.io/docs/concepts/metric_types/#summary)                               | Measure              | Sketch\* or Exact |
+
+A well-designed view API should not require extensive configuration for typical use cases.
+It should also be expressive enough to support custom aggregations and exposition formats without requiring APM vendors to write custom SDKs.
+In short, it should make easy things easy and hard things possible.
+
+## Internal details
+
+A view is defined by:
+
+- A metric instrument
+- An aggregation
+- An optional set of label keys to preserve
+- An optional unique name
+- An optional description
+
+The aggregation may be configured with options specific to its type.
+For example, a _Histogram_ aggregation may be configured with bucket boundaries, e.g. `{0, 1, 10, 100, 200, 1000, inf}`.
+A _Sketch_ aggregation that estimates order statistics (i.e. quantiles), may be configured with a set of predetermined quantiles, e.g. `{.5, .95, .99, 1.00}`.
+
+Each view has a unique name, which may be automatically determined from the metric instrument and aggregation.
+Implementations should refuse to register two views with the same name.
+
+Note that a view does not describe the type (e.g. `int`, `float`) or unit of measurement (e.g. "bytes", "milliseconds") of the metric to be exported.
+The unit is determined by the metric instrument, and the aggregation and exporter may preserve or change the unit.
```

Comment:
It seems like there's only two operations which are appropriate for a unit modification on a metric instrument:

1. preserve the unit
2. change the unit entirely (e.g. to count)

I feel like it may be valuable to call out that there won't be a modification such as "seconds" to "milliseconds". But maybe that's just obvious.